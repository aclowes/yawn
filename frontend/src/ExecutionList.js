import React from 'react';
import {Table, Alert, Button} from 'react-bootstrap';
import {Link} from 'react-router';

import API from "./API";

export default class ExecutionList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {executions: null, error: null};
  }

  componentDidMount() {
    API.get(`/api/executions/`, (payload, error) => {
      this.setState({executions: payload, error});
    });
  }

  renderRows() {
    if (this.state.executions === null) {
      return (
        <tr>
          <td colSpan="3">Loading...</td>
        </tr>
      )
    } else {
      return this.state.executions.map((execution) => (
        <tr key={execution.id}>
          <td><Link to={`/workers/${execution.worker.id}`}>{execution.worker.name}</Link></td>
          <td>{execution.task.workflow.name}</td>
          <td><Link to={`/tasks/${execution.task.id}`}>{execution.task.name}</Link></td>
          <td>{execution.status}</td>
          <td>{execution.start_timestamp}</td>
          <td>{execution.minutes_running}</td>
        </tr>
      ))
    }
  }

  render() {
    return (
      <div>
        <h3>Executions</h3>
        {this.state.error && <Alert bsStyle="danger">{this.state.error}</Alert>}
        <Table striped bordered condensed hover>
          <thead>
          <tr>
            <th>Worker</th>
            <th>Workflow</th>
            <th>Task</th>
            <th>Status</th>
            <th>Start Time</th>
            <th>Minutes Running</th>
          </tr>
          </thead>
          <tbody>
          {this.renderRows()}
          </tbody>
        </Table>
      </div>
    )
  }
}