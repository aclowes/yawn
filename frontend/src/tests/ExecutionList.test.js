import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import ExecutionList from '../ExecutionList';
import {mockAPI} from './mocks'

it('ExecutionList loading', () => {
  mockAPI([null]);
  const component = ReactTestRenderer.create(<ExecutionList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('ExecutionList failure', () => {
  mockAPI();
  const component = ReactTestRenderer.create(<ExecutionList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('ExecutionList success', () => {
  mockAPI([[{
    "id": 17,
    "task": {
      "id": 12,
      "name": "done",
      "workflow": {"id": 1, "name": "Simple Workflow Example", "version": 1},
      "status": "succeeded"
    },
    "worker": {
      "id": 8,
      "name": "graybeard.local 36594",
      "status": "exited",
      "start_timestamp": "2017-01-30T04:17:44.837597Z",
      "last_heartbeat": "2017-02-02T20:10:06.107192Z"
    },
    "status": "succeeded",
    "start_timestamp": "2017-01-30T04:17:47.008524Z",
    "minutes_running": "0m 0.05s"
  }]]);
  const component = ReactTestRenderer.create(<ExecutionList />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
