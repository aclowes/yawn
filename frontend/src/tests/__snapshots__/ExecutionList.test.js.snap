// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExecutionList failure 1`] = `
<div>
  <h3>
    Executions
  </h3>
  <div
    className="alert alert-danger"
    role="alert"
  >
    some error message
  </div>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Worker
        </th>
        <th>
          Workflow
        </th>
        <th>
          Task
        </th>
        <th>
          Status
        </th>
        <th>
          Start Time
        </th>
        <th>
          Runtime
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="6"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ExecutionList loading 1`] = `
<div>
  <h3>
    Executions
  </h3>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Worker
        </th>
        <th>
          Workflow
        </th>
        <th>
          Task
        </th>
        <th>
          Status
        </th>
        <th>
          Start Time
        </th>
        <th>
          Runtime
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="6"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ExecutionList success 1`] = `
<div>
  <h3>
    Executions
  </h3>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          Worker
        </th>
        <th>
          Workflow
        </th>
        <th>
          Task
        </th>
        <th>
          Status
        </th>
        <th>
          Start Time
        </th>
        <th>
          Runtime
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            graybeard.local 36594
          </a>
        </td>
        <td>
          Simple Workflow Example
        </td>
        <td>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            done
          </a>
        </td>
        <td>
          succeeded
        </td>
        <td>
          2017-01-30T04:17:47.008524Z
        </td>
        <td>
          0m 0.05s
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
